src/App.tsx

```diff
@@ -1,27 +1,96 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [error, setError] = useState('');
+  const [loading, setLoading] = useState({
+    all: false,
+    first5: false,
+    red: false,
+  });
+  const errorMessage = 'No goods yet';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const loadAllGoods = async () => {
+    setLoading({ ...loading, all: true });
+    setError('');
+    try {
+      const goodsFromServer = await getAll();
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+      setGoods(goodsFromServer);
+    } catch {
+      setError(errorMessage);
+    }
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    setLoading({ ...loading, all: false });
+  };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const load5Goods = async () => {
+    setLoading({ ...loading, first5: true });
+    setError('');
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    try {
+      const goodsFromServer = await get5First();
+
+      setGoods(goodsFromServer);
+    } catch {
+      setError(errorMessage);
+    }
+
+    setLoading({ ...loading, first5: false });
+  };
+
+  const loadRedGoods = async () => {
+    setLoading({ ...loading, red: true });
+    setError('');
+
+    try {
+      const goodsFromServer = await getRedGoods();
+
+      setGoods(goodsFromServer);
+    } catch {
+      setError(errorMessage);
+    }
+
+    setLoading({ ...loading, red: false });
+  };
+
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={loadAllGoods}
+      >
+        {!loading.all ? 'Load all goods' : 'loadind...'}
+      </button>
+
+      <button
+        type="button"
+        data-cy="first-five-button"
+        onClick={load5Goods}
+      >
+        {!loading.first5 ? 'Load 5 first goods' : 'loadind...'}
+      </button>
+
+      <button
+        type="button"
+        data-cy="red-button"
+        onClick={loadRedGoods}
+      >
+        {!loading.red ? 'Load red goods' : 'loadind...'}
+      </button>
+
+      {!error ? (
+        <GoodsList goods={goods} />
+      ) : (
+        <h2 style={{ color: 'red' }}>No goods yet</h2>
+      )}
```

Comment:
```suggestion
      {!error 
        ? <GoodsList goods={goods} />
        : <h2 style={{ color: 'red' }}>No goods yet</h2>
      }
```
Also, it is better check for error, not its absence